packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,246 @@
+import { referenceForModel } from "../../utils";
+import { OperandStatus } from "../../utils";
+import { getName, getProvider } from "../../utils/selectors";
+import "./list-page.scss";
+
+const tableColumnInfo = [
+  { className: "", id: "name" },
+  {
+    className: "",
+    id: "status"
+  },
+  {
+    className: "",
+    id: "provider"
+  },
+  {
+    className: "",
+    id: "buckets"
+  },
+  { className: "dropdown-kebab-pf pf-c-table__action", id: "" }
+];
+
+type ResourceTableProps = {
+  data: K8sResourceCommon[];
+  unfilteredData: K8sResourceCommon[];
+  loaded: boolean;
+  loadError: any;
+  rowData: CustomData;
+};
+
+const ResourceTable: React.FC<ResourceTableProps> = (props) => {
+  const { t } = useTranslation();
+  const tableColumns = React.useMemo<TableColumn<K8sResourceCommon>[]>(
+    () => [
+      {
+        title: t("Name"),
+        sort: "metadata.name",
+        transforms: [sortable],
+        props: {
+          className: tableColumnInfo[0].className
+        },
+        id: tableColumnInfo[0].id
+      },
+      {
+        title: t("Status"),
+        props: {
+          className: tableColumnInfo[1].className
+        },
+        id: tableColumnInfo[1].id
+      },
+      {
+        title: t("Provider"),
+        props: {
+          className: tableColumnInfo[2].className
+        },
+        id: tableColumnInfo[2].id
+      },
+      {
+        title: t("Buckets"),
+        props: {
+          className: tableColumnInfo[3].className
+        },
+        id: tableColumnInfo[3].id
+      },
+      {
+        title: "",
+        props: {
+          className: tableColumnInfo[4].className
+        },
+        id: tableColumnInfo[4].id
+      }
+    ],
+    [t]
+  );
+
+  const [columns] = useActiveColumns({
+    columns: tableColumns,
+    showNamespaceOverride: false,
+    columnManagementID: null
+  });
+  return (
+    <VirtualizedTable
+      {...props}
+      aria-label={t("Resource Page")}
+      columns={columns}
+      Row={RowRenderer}
+    />
+  );
+};
+
+type CustomData = {
+  launchModal: LaunchModal;
+  resourceModel: K8sModel;
+  kebabActions?: (t: TFunction) => { [key: string]: { value: string } };
+};
+
+const RowRenderer: React.FC<RowProps<K8sResourceCommon, CustomData>> = ({
+  obj,
+  activeColumnIDs,
+  rowData
+}) => {
+  const { resourceModel, launchModal, kebabActions } = rowData;
+  const name = getName(obj);
+  const provider = getProvider(obj);
+  return (
+    <>
+      <TableData {...tableColumnInfo[0]} activeColumnIDs={activeColumnIDs}>
+        {name}
+      </TableData>
+      <TableData {...tableColumnInfo[1]} activeColumnIDs={activeColumnIDs}>
+        <OperandStatus operand={obj} />
+      </TableData>
+      <TableData {...tableColumnInfo[2]} activeColumnIDs={activeColumnIDs}>
+        {provider}
+      </TableData>
+      <TableData {...tableColumnInfo[3]} activeColumnIDs={activeColumnIDs}>
+        {2 + " Buckets"}
+      </TableData>
```

Comment:
add a `ToDo` here, so that we remember to correct it... it is hardcoded right now...
or just fix it straight away, why have we hardcoded "2 Buckets", we already have APIs to get this info...

Reply:
I will suggest we get the count from the API then just the pop-over/redirect will be left.

Reply:
Done